docs/decisions/2022-07-pet-test-fixtures.md

```diff
@@ -0,0 +1,73 @@
+# Allow "pet" test fixtures
+
+* **Status:** approved
+* **Last Updated:** 2022-07
+
+## Context & Problem Statement
+
+Emblem evolved to use "pet" test fixtures in end-to-end (E2E) test runs. These include resources such as Google Cloud projects, Cloud Storage buckets, container images in Artifact Registry, and Cloud Run services.
+
+We decided to investigate whether we should migrate our fixtures to a "cattle"-based model.
+
+## Priorities
+
+Ideally, our test fixtures would be designed such that:
+
+* Test infrastructure is quick to set up.
+* Test infrastructure is documented (e.g. via Terraform files).
+* Tests themselves are quick to write.
+* Both tests and their infrastructure require minimal maintenance.
+* Tests are runnable by multiple [teams of] developers in parallel.
+* Tests cover as much of the application surface as possible.
+* Tests allow developers and contributors to quickly identify root causes of bugs.
+
+Of course, some of these goals are mutually exclusive: supporting parallel test runs often increases the cost of writing (and in some cases maintaining) tests and test infrastructure.
+
+## Constraints
+
+### Some resources MUST be pets
+While most of Emblem's Google Cloud resources are provisioned programmatically using Terraform, some (such as the Google Cloud projects themselves) are provisioned manually.
+
+> Consequently, we can't rely entirely on automated provisioning systems (at least, not without significant further engineering work) to spin these up and down arbitrarily.
+
+### Emblem is owned by a single team
+While Emblem is designed with a service-oriented architecture, all of Emblem's components are (currently) owned by the same team.
+
+> Consequently, optimizing our test fixtures for multi-team collaboration isn't as important as minimizing test development/maintenance costs.
```

Comment:
I'm not sure I follow this. Is the idea that in a multi-team scenario, additional automation of test fixtures is more reasonable to avoid duplicating work? If so I think this needs to be more explicit.